### PR TITLE
Migrate the Glean SDK to the Rust implementation

### PR DESCRIPTION
**Pull request by Dexterp37**
This completely removes the Kotlin implementation of the Glean SDK from this repository. It introduces aliases that are used instead.

Please note that integration tests keep living in this repository as part of the [Glean sample app](https://github.com/mozilla-mobile/android-components/tree/master/samples/glean/src/androidTest/java/org/mozilla/samples/glean).

**Note: this is a massive PR with deletion, mostly. I'd recommend reviewing it commit-by-commit**

**TODO**

- [x] remove the glean binary;
- [x] depend on the maven version of the Glean SDK.

---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt [hunk 1]
```diff
@@ -4,642 +4,4 @@
...
-object Glean : GleanInternalAPI() {
-    internal const val SCHEMA_VERSION = 1
-
-    /**
-     * The name of the directory, inside the application's directory,
-     * in which Glean files are stored.
-     */
-    internal const val GLEAN_DATA_DIR = "glean_data"
-}
+typealias Glean = mozilla.telemetry.glean.Glean
```
[the remark this answers is not in the document]
Comment (Dexterp37) [reply components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt]: The uploader itself is changed through the `Configuration` object that is passed through the `Glean.initialize` method, so the `Glean` object itself can be safely typealiased

[the remark this answers is not in the document]
Comment (Dexterp37) [reply components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt]: oh, actually, you're right! We need to do that :) But not here: in the `Configuration.kt` object itself :)

[the remark this answers is not in the document]
Comment (mdboom) [reply components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt]: Does that mean we should have a `Configuration` object with a different default here?  I don't think Fenix sets it explicitly for example, and we should "do the right thing" by default when people get Glean through android-components, which is probably to use the concept fetch one.

**Comment by Dexterp37** [comment]
> 
> 
> Is adding the binary `glean-0.0.1-TESTING1.aar` to source control intentional?

Yes, it's meant to be removed after the library is available on Maven (see the TODO in the PR description)

File: components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dexterp37) [reply components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt]: Mh, good point. I think this requires changes to both files to retain the current behaviour.

**Comment by travis79** [comment]
> > Is adding the binary `glean-0.0.1-TESTING1.aar` to source control intentional?
> 
> Yes, it's meant to be removed after the library is available on Maven (see the TODO in the PR description)

Sorry, I totally overlooked the TODO. This makes sense, I just thought it was weird to include a binary and wanted to make sure.

**Comment by Dexterp37** [comment]
bors=mdboom,travis79,badboy
bors r+

**Pull request merged**